### DISCUSSION TITLE
When did ottava ( 8va 15va etc. ) on tab get fixed?

### DISCUSSION THREAD

**Discussion by mkj42** [musescore/MuseScore#24561] [Q&A] [answered]
I have been searching github and through google, and I cannot locate when and how this got fixed.
It has clearly been solved since at least a 4.4 nightly:
OS: Windows 10 Version 2009 or later, Arch.: x86_64, MuseScore Studio version (64-bit): 4.4.0-242210305, revision: github-musescore-musescore-23820f3

In 3.7 and earlier, in any stringed instrument (?, I haven't checked them all), if you add an ottava line for transposition, 2 incorrect things happen:

1.    The ottava line is visible on the tab staff; it should only be visible on the standard staff.
2.    The notes do not change displayed fret/string number on the tab staff; they should. They sound at the correct new pitch, but are still written at the original octave.

![tab 8va error](https://github.com/user-attachments/assets/188c6b56-a6be-4b30-b737-d896d85bad03)

I am sincerely hoping to target the code changes that fixed this so it could possibly be backported to 3.7.
https://github.com/Jojo-Schmitz/MuseScore/issues/572

Thanks

**Comment by cbjeukendrup** [comment, marked as answer]
This seems related: https://github.com/musescore/MuseScore/pull/11319

**Reply by cbjeukendrup** [reply]
And if it turns out that that's not all, https://github.com/musescore/MuseScore/pull/10765 might also play a role, given that it does something with `pitchOffset`

**Reply by mkj42** [reply]
Thanks much Casper, though I don't know for sure yet whether to mark either of these as the answer.
Please give me a bit.

**Reply by Jojo-Schmitz** [reply]
Spot on! Both of them are needed, backport in progress... see #622

**Comment by Jojo-Schmitz** [comment]
That'd mean though that in Mu4 the issue got fixed much earlier than for 4.4, basically ever since 4.0

**Reply by mkj42** [reply]
I only noticed it at 4.4.  I had barely tried any 4s until 4.3.2
  sorry if my mentioning 4.4 was misleading.

**Reply by Jojo-Schmitz** [reply]
4.1.0 had it fixed already apparently, haven't yet tested anything older

**Comment by Jojo-Schmitz** [comment]
Just found another bug with ottavas (in 3.x and 4.x):
![image](https://github.com/user-attachments/assets/c2a010b8-354c-4467-af77-ae42074b644c)
It is possible to attach more than one ottava at a time. This shouldn't happen, or should it?

**Reply by mkj42** [reply]
Unclear
Did trying to enter 1 result in both?
I believe I saw that in the forum.

**Reply by Jojo-Schmitz** [reply]
Enter one, then the other. And yet another etc. They just accumulate.
There should be at most one (per voice?)

**Reply by mkj42** [reply]
Still unclear
Do they accumulate in the current 3.7?
If so, how are you doing that?
It seems fine on my end.
I even tried selecting an entire measure first - still fine.

**Reply by Jojo-Schmitz** [reply]
In 3.x and 4.x. Select a measure, apply an ottava. Select the same measure again (if if didn't stay selected) and apply another ottava. Leather, rinse, repeat, ad lib.

**Reply by mkj42** [reply]
Gotcha
I never tried that before.
Strange, but not too big a bug to work around.
Backporting the original problem will be wonderful.  Thank you.
